reloc.ml

```diff
@@ -184,28 +184,17 @@ let build_diversion lst =
       ) (List.filter (fun f -> f <> "") lst)
   in
   let utf16, lst =
-    try true, List.map toutf16 lst
-    with Not_utf8 -> false, lst
+    match !toolchain with
+    | `MSVC | `MSVC64 | `LIGHTLD ->
+        (try (true, List.map toutf16 lst) with Not_utf8 -> (false, lst))
+    | _ -> (false, lst)
   in
   if utf16 then output_string oc "\xFF\xFE"; (* LE BOM *)
```

Comment:
I'd prefer to have the options fully enumerated (as was on L201-202) below, but overall I wonder if this would be clearer for the future:
```suggestion
    match !toolchain with
    | `MINGW | `MINGW64 | `GNAT | `GNAT64 | `CYGWIN64 -> lst
    | `MSVC | `MSVC64 | `LIGHTLD ->
        (* UTF-16 response files required *)
        try
          let lst = List.map toutf16 lst in
          output_string oc "\xFF\xFE"; (* LE BOM *)
          lst
        with Not_utf8 -> lst
  in
```